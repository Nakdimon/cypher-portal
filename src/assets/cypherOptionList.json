[
    {
        "label": "Casear",
        "value": "caesar"
    },
    {
        "label": "ROT13",
        "value": "rot13"
    }
]