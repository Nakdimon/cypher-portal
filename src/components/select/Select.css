.cypher-select {
    height: 35px;
    margin: 15px 0px;
    border-radius: 5px; 
  }