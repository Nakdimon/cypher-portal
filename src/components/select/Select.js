import React from 'react';
import './Select.css'

function Select({
    cypherOptionList,
    handleSelectChange
}) {
    return (
        <select className="cypher-select" defaultValue={'DEFAULT'} onChange={handleSelectChange}>
          <option disabled value="DEFAULT">Select a cypher</option>
          {
            cypherOptionList.map((cypherOption) => 
            <option className="cypher-option" key={cypherOption.value} value={cypherOption.value}>{cypherOption.label}</option>)
          }
        </select>
    )

}

export default Select;