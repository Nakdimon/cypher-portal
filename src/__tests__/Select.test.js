import React from 'react';
import { create } from 'react-test-renderer';
import Select from '../components/select/Select';
import cypherOptionList from '../assets/cypherOptionList.json'

const handleSelectChange = () => 'dummyHandleSelectChange'

describe("Select component: ", ()=> {
  it("Matches the snapshot", () => {
    const select = create(
      <Select
        cypherOptionList={cypherOptionList} 
        handleSelectChange={handleSelectChange}
      />
    )
    expect(select).toMatchSnapshot();
  })
})
