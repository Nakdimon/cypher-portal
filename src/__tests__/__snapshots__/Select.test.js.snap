// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App component:  Matches the snapshot 1`] = `
<select
  className="cypher-select"
  onChange={[Function]}
>
  <option
    disabled={true}
    selected={true}
  >
    Select a cypher
  </option>
  <option
    className="cypher-option"
    value="caesar"
  >
    Casear
  </option>
  <option
    className="cypher-option"
    value="rot13"
  >
    ROT13
  </option>
</select>
`;

exports[`Select component:  Matches the snapshot 1`] = `
<select
  className="cypher-select"
  onChange={[Function]}
>
  <option
    disabled={true}
    selected={true}
  >
    Select a cypher
  </option>
  <option
    className="cypher-option"
    value="caesar"
  >
    Casear
  </option>
  <option
    className="cypher-option"
    value="rot13"
  >
    ROT13
  </option>
</select>
`;
